File: src/kernels/transport/labos/basis.zig
//! Purpose:
//!   Thin LABOS basis facade over shared types, matrix algebra, and phase basis logic.

const types = @import("types.zig");
const matrix = @import("matrix.zig");
const phase_basis = @import("phase_basis.zig");

pub const max_gauss = types.max_gauss;
pub const max_extra = types.max_extra;
pub const max_nmutot = types.max_nmutot;
pub const max_n2 = types.max_n2;
pub const max_phase_coef = types.max_phase_coef;

pub const Mat = types.Mat;
pub const Vec = types.Vec;
pub const Vec2 = types.Vec2;
pub const LayerRT = types.LayerRT;
pub const UDField = types.UDField;
pub const UDLocal = types.UDLocal;
pub const Geometry = types.Geometry;

pub const smul = matrix.smul;
pub const esmul = matrix.esmul;
pub const semul = matrix.semul;
pub const matAdd = matrix.matAdd;
pub const qseries = matrix.qseries;

pub const PhaseKernel = phase_basis.PhaseKernel;
pub const FourierPlmBasis = phase_basis.FourierPlmBasis;
pub const fillZplusZminFromBasis = phase_basis.fillZplusZminFromBasis;
pub const fillZplusZminFromBasisLimited = phase_basis.fillZplusZminFromBasisLimited;
pub const fillZplusZmin = phase_basis.fillZplusZmin;

File: rust_src/forward_model/radiative_transfer/labos/basis.rs
pub use super::attenuation::{
    AttenArray, DynamicAttenArray, fill_attenuation, fill_attenuation_dynamic,
    fill_attenuation_dynamic_with_grid,
};
pub use super::matrix::{
    esmul, esmul_semul, esmul_semul_add, mat_add, mat_add_esmul, mat_add_esmul3, mat_add_semul3,
    qseries, qseries_known_nonzero_product, semul, semul_add, smul, smul_add_semul3,
    smul_add_semul3_known_right_trace, smul_into, smul_known_traces, smul_known_traces_if_nonzero,
};
pub use super::orders::{
    AttenuationLookup, OrdersResult, OrdersResultView, OrdersWorkspace, dot_gauss,
    orders_scat_into, orders_scat_into_with_active, orders_scat_into_with_active_local_sum,
    orders_scat_into_with_local_sum, orders_scat_transport_into,
};
pub use super::phase_basis::{
    FourierPlmBasis, PhaseKernel, PhaseKernelRow, fill_zplus_zmin, fill_zplus_zmin_from_basis,
    fill_zplus_zmin_from_basis_limited, fill_zplus_zmin_row_from_basis_limited,
};
pub use super::reflectance::{
    calc_integrated_reflectance, calc_integrated_reflectance_with_basis, calc_reflectance,
    calc_reflectance_tangent, fill_adjacent_layer_phase_max_indices, resolved_fourier_max,
    resolved_phase_coefficient_max, total_scattering_optical_depth,
};
pub use super::types::{
    Geometry, LayerRt, MAX_EXTRA, MAX_GAUSS, MAX_N2, MAX_NMUTOT, MAX_PHASE_COEF, Mat, UdField,
    UdLocal, Vec, Vec2,
};
pub use super::workspace::{
    GeometryCacheStatus, LayerPhaseSignatureProbe, PlmBasisCacheStatus, Workspace,
};
